domain/secret/state/state.go

```diff
@@ -431,8 +432,46 @@ FROM secret
 	return dbSecrets.toSecretMetadata(dbsecretOwners)
 }
 
+// GetUserSecretURIByLabel returns the URI for the user secret with the specified label,
+// or an error satisfying [secreterrors.SecretNotFound] if the secret does not exist.
+func (st State) GetUserSecretURIByLabel(ctx context.Context, label string) (*coresecrets.URI, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	query := `	
+SELECT id AS &secretInfo.id
+FROM   secret
+JOIN   secret_model_owner mso ON secret.id = mso.secret_id
+WHERE  mso.label = $M.label
+	`
+
+	queryStmt, err := st.Prepare(query, secretInfo{}, sqlair.M{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var dbSecrets secrets
```

Comment:
Is it `secretInfo` or `secrets`?

Reply:
secrets is just the type alias name. I found it nicer that secretInfos